[MTM-61220] Java SDK improved error propagation for authenticated req… (#4673)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2026-34-0-java-sdk-error-propagation-authencitaion-request.md

---
date: ''
title: Java SDK improved error propagation for authenticated requests
product_area: Application enablement & solutions
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: QWPx3rFfn
    label: Java SDK
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-61220
version: 2026.34.0
---

> Fixed an issue where unexpected exceptions during token-authenticated service calls were incorrectly mapped to 401 Unauthorized. The SDK now correctly propagates the original error status and metadata, ensuring accurate debugging and preventing misleading authentication alerts.

Reviewer: ```suggestion
Previously, unexpected exceptions during token-authenticated service calls were incorrectly mapped to "401 Unauthorized". This issue has been fixed. The Java SDK now correctly propagates the original error status and metadata, ensuring accurate debugging and preventing misleading authentication alerts.
```
